### ISSUE TITLE
[bitnami/opencart] Unable to login at *IP_address*/admin

### ISSUE THREAD

**Issue by marangiop** [bitnami/bitnami-docker-opencart#81]
**Description**

I do not know what to put down as username and password at 62.14.219.13/admin. I tried "demo" for both, but then the page crashes. I tried bn_opencart (which is the OPENCART_DATABASE_USER inside docker-compose.yml) and no password, but again the page crashes. My goal is to install an Opencart plugin, but first I need to be able to login into admin.

**Steps to reproduce the issue:**

1. Run container using docker compose yml file:
2. Set openccart host inside docker compose yml file and http/https server inside config.php (inside opencart container) to desired IP address
3. Stop and restart both mariadb and opencart containers

**Describe the results you received:**

Admin page is not functional.

**Describe the results you expected:**

I expect to be able to log in and install opencart plugins.

**Version**

- Output of `docker version`:

Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:23:03 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:25:29 2018
  OS/Arch:          linux/amd64
  Experimental:     false

- Output of `docker info`:

Containers: 6
 Running: 2
 Paused: 0
 Stopped: 4
Images: 58
Server Version: 18.06.1-ce
Storage Driver: devicemapper
 Pool Name: docker-253:3-1680187339-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Udev Sync Supported: true
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data loop file: /home/docker/devicemapper/devicemapper/data
 Metadata loop file: /home/docker/devicemapper/devicemapper/metadata
 Data Space Used: 16.01GB
 Data Space Total: 107.4GB
 Data Space Available: 91.37GB
 Metadata Space Used: 18.73MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.129GB
 Thin Pool Minimum Free Space: 10.74GB
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.146-RHEL7 (2018-01-22)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-862.11.6.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 31.2GiB
Name: supersede.es.atos.net
ID: HO72:KJ3E:WCTD:DLUO:BG7L:6EHU:S7LQ:PYNT:BOEX:CJ3X:FGJY:EJMW
Docker Root Dir: /home/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

- Output of `docker-compose version` (if applicable):

docker-compose version 1.22.0, build f46880fe
docker-py version: 3.4.1
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.1.0f  25 May 2017

**Comment by dani8art** [comment]
Hi, @marangiop [here](https://github.com/bitnami/bitnami-docker-opencart/blob/master/3/debian-10/Dockerfile#L47) you can see the default username and password for the container. Also, you can see other environment variables that could be very useful.

**Comment by marangiop** [comment]
HI @dani8art , I have tried to type in the username and password according to the link you suggested, but as before, right after entering the credentials the pages crashes. Any ideas as to why this may be happening?

**Comment by dani8art** [comment]
Ok so I think there could be something wrong , could you try to run using this environment var and then attach your logs here, please? 

```
NAMI_DEBUG="--log-level=debug"
```

**Comment by marangiop** [comment]
I have deleted the older containers, then added the line you suggested inside the docker-compose.yml file and re-deployed

Here are the logs

[paolo.marangio@supersede test2]$ sudo docker logs b094ef3728c9
app-entrypoint.sh 15:20:33.96
app-entrypoint.sh 15:20:33.97 Welcome to the Bitnami opencart container
app-entrypoint.sh 15:20:33.97 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-opencart
app-entrypoint.sh 15:20:33.97 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-opencart/issues
app-entrypoint.sh 15:20:33.97
app-entrypoint.sh 15:20:33.99 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
nami    INFO  Initializing apache
nami    INFO  apache successfully initialized
nami    INFO  Initializing php
php     TRACE [configurePermissions] List of files: ["/opt/bitnami/php/tmp"]
php     TRACE [configurePermissions] File to chown: /opt/bitnami/php/tmp
php     TRACE [configurePermissions] List of files: ["/opt/bitnami/php/logs"]
php     TRACE [configurePermissions] File to chown: /opt/bitnami/php/logs
php     TRACE [configurePermissions] List of files: ["/opt/bitnami/php/var","/opt/bitnami/php/var/log","/opt/bitnami/php/var/run"]
php     TRACE [configurePermissions] File to chown: /opt/bitnami/php/var
php     TRACE [configurePermissions] File to chown: /opt/bitnami/php/var/log
php     TRACE [configurePermissions] File to chown: /opt/bitnami/php/var/run
nami    INFO  php successfully initialized
nami    INFO  Initializing mysql-client
nami    INFO  mysql-client successfully initialized
nami    INFO  Initializing opencart
apache  TRACE Skipping reload: com.bitnami.apache not running
apache  TRACE [runProgram] Executing: null /opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf
apache  TRACE code: 0
apache  TRACE stdout:

apache  TRACE stderr:

opencar INFO  Preparing webserver environment...
apache  TRACE [runProgram] Executing: /opt/bitnami/apache/bin/apachectl graceful
opencar INFO  Preparing PHP environment...
mysql-c INFO  Trying to connect to MySQL server
mysql-c INFO  Found MySQL server listening at mariadb:3306
mysql-c TRACE [canConnect] Testing connection with 'bn_opencart' user
mysql-c TRACE [showDatabases] Getting databases for 'bn_opencart' user
mysql-c TRACE [execute] Executing: mysql -N -P 3306 -hmariadb -ubn_opencart -e SHOW DATABASES;
mysql-c TRACE [execute] Result: {"code":0,"stdout":"bitnami_opencart\ninformation_schema\ntest\n","stderr":""}
mysql-c TRACE [showDatabases] Databases: bitnami_opencart
information_schema
test

mysql-c TRACE [canConnect] Connection with 'bn_opencart' user is successful
mysql-c INFO  MySQL server listening and working at mariadb:3306
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/config.php
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/config.php
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/view/stylesheet/bootstrap.css
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/system/storage
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/image
opencar INFO  Installing OpenCart...
php     TRACE [runProgram] Executing: /opt/bitnami/php/bin/php cli_install.php,install,--db_hostname,mariadb,--db_port,3306,--db_username,bn_opencart,--db_password,,--db_database,bitnami_opencart,--db_driver,mysqli,--username,user,--password,bitnami1,--email,user@example.com,--http_server,http://62.14.219.13/
opencar INFO  Configuring SMTP...
opencar TRACE [prepareDataToPersist] Preparing /opt/bitnami/opencart/config.php to persist in /bitnami/opencart/config.php
opencar TRACE [prepareDataToPersist] Preparing /opt/bitnami/opencart/admin/config.php to persist in /bitnami/opencart/admin/config.php
opencar TRACE [prepareDataToPersist] Preparing /opt/bitnami/opencart/admin/view/stylesheet/bootstrap.css to persist in /bitnami/opencart/admin/view/stylesheet/bootstrap.css
opencar TRACE [prepareDataToPersist] Preparing /opt/bitnami/opencart/image to persist in /bitnami/opencart/image
opencar TRACE [prepareDataToPersist] Preparing /opt/bitnami/opencart/system/storage to persist in /bitnami/opencart/system/storage
apache  TRACE [runProgram] Executing: /opt/bitnami/apache/bin/apachectl graceful
opencar INFO  Configuring permissions
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/controller
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/language
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/model
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/admin/view
opencar TRACE [configurePermissions] File to chown: /opt/bitnami/opencart/catalog
opencar TRACE [configurePermissions] File to chown: /bitnami/opencart/image
opencar TRACE [configurePermissions] File to chown: /bitnami/opencart/system/storage
opencar TRACE [configurePermissions] File to chown: /bitnami/opencart/config.php
opencar TRACE [configurePermissions] File to chown: /bitnami/opencart/admin/config.php
apache  TRACE [runProgram] Executing: /opt/bitnami/apache/bin/apachectl graceful
opencar INFO
opencar INFO  ########################################################################
opencar INFO   Installation parameters for opencart:
opencar INFO     Username: user
opencar INFO     Password: **********
opencar INFO     Email: user@example.com
opencar INFO     Site URL: http://62.14.219.13/
opencar INFO   (Passwords are not shown for security reasons)
opencar INFO  ########################################################################
opencar INFO
nami    INFO  opencart successfully initialized
app-entrypoint.sh 15:21:00.36 INFO  ==> Starting gosu...
[Mon May 11 15:21:00.479216 2020] [ssl:warn] [pid 132] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Mon May 11 15:21:00.480207 2020] [ssl:warn] [pid 132] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Mon May 11 15:21:00.536303 2020] [ssl:warn] [pid 132] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Mon May 11 15:21:00.537123 2020] [ssl:warn] [pid 132] AH01909: www.example.com:443:0 server certificate does NOT include an ID which matches the server name
[Mon May 11 15:21:00.557733 2020] [mpm_prefork:notice] [pid 132] AH00163: Apache/2.4.43 (Unix) OpenSSL/1.1.1d PHP/7.3.17 configured -- resuming normal operations
[Mon May 11 15:21:00.557781 2020] [core:notice] [pid 132] AH00094: Command line: 'httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'

**Comment by dani8art** [comment]
Hi @marangiop,  I was trying in my environment and the error that I received is an HTTPS redirection problem/misconfiguration. When I do a login action I'm redirected to the HTTPS endpoint. I mean, I'm doing login over: 

```
http://localhost:8080/admin/index.php?route=common/login
```

when I click on login, it redirects me to:

```
https://localhost:4443/admin/index.php?route=common/login
```

That is the https endpoint for my deployment. 

There is two way of tackling that: 
- Disabling HTTPS redirections.
- Configuring HTTPS in a way that works.

1. Disabling HTTPS redirections. [This](https://forum.opencart.com/viewtopic.php?t=30750#p150234) could help to solve this issue. 

2. Configuring HTTPS in a way that works. you should be sure that the port for HTTPS is open 443 in that case. 

PD. If you could share a bit more about the error itself, it would be very useful.

**Comment by marangiop** [comment]
Hi @dani8art , apologies for not replying.

I managed to get it working using the Bitnami Opencart Chart (https://github.com/bitnami/charts/tree/master/bitnami/opencart/)

My end goal was to deploy OpenCart on K8S, so using the Chart made more sense. I was just testing the bitnami Docker-compose workflow because I was having issues with installing the Chart as well, but I solved those so now it´s all good.
